File: task_equivalents/llm_translations/llama/base/python/data_visualization/01_scatter_trend_grouped.py
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression

# Load the data
auto = pd.read_csv('data/csv/Auto.csv', na_values='?').dropna()

# Prepare the data
X = auto[['horsepower']].values
Y = auto['mpg'].values.reshape(-1, 1)

# Add a constant (intercept) to the independent value
X_const = np.hstack((np.ones((X.shape[0], 1)), X))

# Perform simple linear regression
model = LinearRegression(fit_intercept=False)
model.fit(X_const, Y)

a, b = model.coef_[0]
yfit = model.predict(X_const)

# Plot the data and the regression line
plt.scatter(X, Y)
plt.plot(X, yfit, color='red')
plt.xlabel('horsepower')
plt.ylabel('mpg')
plt.show()

File: task_equivalents/llm_translations/llama/base/r/data_visualization/01_scatter_trend_grouped.R
library(readr)
library(ggplot2)

Auto <- read_csv('data/csv/Auto.csv', na = "?", col_types = cols(.default = "d")) %>%
  drop_na()

X <- Auto$horsepower
Y <- Auto$mpg

model1_fit <- lm(Y ~ X)

a <- coef(model1_fit)[1]
b <- coef(model1_fit)[2]
yfit <- a + b * X

ggplot(Auto, aes(x = X, y = Y)) +
  geom_point() +
  geom_line(aes(x = X, y = yfit), color = "red") +
  labs(x = "horsepower", y = "mpg")